docs/messaging/messaging-spans.md

```diff
@@ -261,6 +261,23 @@ batch of messages, or for no message at all (if it is signalled that no
 messages were received). For each message it accounts for, the "Process" or
 "Receive" span SHOULD link to the message's creation context.
 
+> [!IMPORTANT]
+> These conventions use spans links as the default mechanism to correlate
+> producers and consumer(s) because:
+>
+> - It is the only consistent trace structure that can be guaranteed,
+> given the many different messaging systems models available.
+>
+> - It is the only option to correlate produce and consumer(s) in batch scenarios
+> as a span can only have a single parent.
+>
+> - Offers a consistent experience for users analysing traces from different
+> messaging systems.
+
+Exclusively for single messages scenarios, "Process" or "Receive" span MAY
+use the message's creation context as its parent, thus achieving a direct
+parent-child relationship between producer and consumer(s).
```

Comment:
I'm suggesting to add a bit of additional guidance around the ambient context and defaults.

```suggestion
parent-child relationship between producer and consumer(s). Instrumentations SHOULD document whether they use message creation context as a parent for "Process" spans and MAY provide configuration option allowing users to control this behavior. 

It's NOT RECOMMENDED to use message creation context as a parent for "Process" span (by default) if processing usually happens in the scope of another span.

If instrumentation uses message creation context as a parent on "Process" span in the scope of another valid ambient context, it SHOULD set the ambient context as a link on "Process" span to preserve correlation between message processing and that context. 

For example, a messaging broker pushes messages to consumers over HTTP and a consumer application has HTTP server and messaging instrumentations enabled. The messaging instrumentation would create "Process" span following one of these possible approaches:

- "Process" span is a child of HTTP server span context and has a link to message creation context. This is the default behavior for this instrumentation.
- "Process" span is a child of message creation context and has two links: one to message creation context and another one to HTTP server span context. This is opt-in behavior.
```

LMK what you think. 

Maybe we can also put it all into a new section like "Using message context as a parent for "Process" spans" so it appears in the TOC?

Reply:
I was thinking the same, but to add new headers for `Receive` spans and `Process` inside the `Consumer spans` section. Right now they are all together and I find it a bit hard to follow.

Reply:
Thank you! I like "Message creation context as parent of "Process" span"!